pkg/skaffold/schema/latest/v1/config.go

```diff
@@ -1208,6 +1208,11 @@ type DockerArtifact struct {
 	// For example: `{"key1": "value1", "key2": "value2"}`.
 	BuildArgs map[string]*string `yaml:"buildArgs,omitempty"`
 
+	// Env are environment variables passed to the docker build.
+	// It also accepts environment variables via the go template syntax.
+	// For example: `[{"name": "key1", "value": "value1"}, {"name": "key2", "value": "value2"}, {"name": "key3", "value": "'{{.ENV_VARIABLE}}'"}]`.
+	Env []v1.EnvVar `yaml:"env,omitempty"`
```

Comment:
Docker doesn't propagate environment fields, so this doesn't really make sense to be included here.

Reply:
The irony is that there's no need to modify the config for Docker artifacts. At least running Skaffold locally, the file `pkg/skaffold/build/gcb/docker.go` is not called at all. Instead, most of the work is done in pkg/skaffold/docker/image.go.  func (l *localDaemon) Build() calls EvalBuildArgs(), which turns out to be just fine; the missing env parameter isn't really well named, since it's used for non-environmental variables that need to be injected into templates.

I learned this by setting up delve with VS Code, and once I put in a .env file into the top skaffold/ directory of the repo, to my shock the templates took up the environment automatically. master / HEAD is not broken in this respect. But outside of the debugger, broken. 

I believe the problem is with how skaffold populates the environment. I can't get .env files to work with skaffold, pretty much no matter where I put them. Do you know how .env files are implemented? I can't find any loader code for the life of me; neither of the two common golang modules used for loading env files is called anywhere, and I could not find any code that was parsing the files and inserting them into the process environment. I'm guessing that VS Code exports the file in code, and then calls fork(), thus passing on the environment pristine. Whatever skaffold is currently doing does not work as well.

Reply:
OK, my theory seems sound. I wrote a script to parse my .env file and exported the variable. Do that, and calling skaffold dev resolves my build arg template. 

So how can we fix the way that skaffold initializes its environment?

Reply:
I added a non-trivial test of the templating feature. The style of the test is a little different from most of the table driven tests used for docker, but was needed to actually exercise `EvalBuildArgs()`.  Short version: the feature works fine in 1.24, and likely earlier than that.

Reply:
This PR meandered too much. I've taken the final version and put it in as [PR-5901](https://github.com/GoogleContainerTools/skaffold/pull/5901), which should be much easier to understand. Leaving this one open for now so people know the background of it, since this is my first issue in the Skaffold repo.